src/gt4py/storage/allocators.py

```diff
@@ -109,29 +135,25 @@ def ndim(self):
         return len(self.shape)
 
     def __array__(self, dtype: Optional[np.dtype] = None) -> np.ndarray:
-        try:
-            return self.ndarray.__array__(dtype=dtype)
-        except (AttributeError, TypeError):
+        if not hasattr(self.ndarray, "__array__"):
```

Comment:
This is fixing some mypy complains, but what was the `TypeError` in the `except`? Checking if dtype works? Maybe then it's fine if we let the lower level error propagate?